File: src/main/java/me/miki/shindo/management/mods/impl/PlayerCounterMod.java
package me.miki.shindo.management.mods.impl;

import me.miki.shindo.management.event.EventTarget;
import me.miki.shindo.management.event.impl.EventRender2D;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.mods.SimpleHUDMod;
import me.miki.shindo.management.nanovg.font.LegacyIcon;

import me.miki.shindo.management.settings.config.Property;
import me.miki.shindo.management.settings.config.PropertyType;
public class PlayerCounterMod extends SimpleHUDMod {

    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ICON)
    private boolean iconSetting = true;

    public PlayerCounterMod() {
        super(TranslateText.PLAYER_COUNTER, TranslateText.PLAYER_COUNTER_DESCRIPTION);
    }

    @EventTarget
    public void onRender2D(EventRender2D event) {
        this.draw();
    }

    @Override
    public String getText() {
        return "Player: " + mc.thePlayer.sendQueue.getPlayerInfoMap().size();
    }

    @Override
    public String getIcon() {
        return iconSetting ? LegacyIcon.USERS : null;
    }
}

File: src/main/kotlin/me/miki/shindo/management/mods/impl/PlayerCounterMod.kt
package me.miki.shindo.management.mods.impl

import me.miki.shindo.management.event.EventTarget
import me.miki.shindo.management.event.impl.EventNVG
import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.mods.SimpleHUDMod
import me.miki.shindo.management.nanovg.font.LegacyIcon
import me.miki.shindo.management.settings.config.Property
import me.miki.shindo.management.settings.config.PropertyType

class PlayerCounterMod : SimpleHUDMod(
    TranslateText.PLAYER_COUNTER,
    TranslateText.PLAYER_COUNTER_DESCRIPTION,
    LegacyIcon.MOD_PLAYER_COUNTER
) {
    @Property(type = PropertyType.BOOLEAN, translate = TranslateText.ICON)
    private val iconSetting = true

    @EventTarget
    fun onRender2D(event: EventNVG) {
        this.draw()
    }

    override fun getText(): String {
        return "Player: " + mc.thePlayer.sendQueue.playerInfoMap.size
    }

    override fun getIcon(): String? {
        return if (iconSetting) LegacyIcon.USERS else null
    }
}


